types/request/index.d.ts

```diff
@@ -134,7 +134,7 @@ declare namespace request {
         jsonReplacer?: (key: string, value: any) => any;
         multipart?: RequestPart[] | Multipart;
         agent?: http.Agent | https.Agent;
-        agentOptions?: SecureContextOptions;
+        agentOptions?: http.AgentOptions | https.AgentOptions | SecureContextOptions;
```

Comment:
This change works for me. 

**Nitpick**: It's a little redundant since `http.AgentOptions` extends `http.AgentOptions` and `tls.ConnectionOptions`. And `tls.ConnectionOptions` extends `tls.SecureContextOptions`.
But personally I think the readability is better if the `http` and `https` are listed.
Maybe narrow it down to just `agentOptions?: http.AgentOptions | https.AgentOptions;`.